### ISSUE TITLE
LET: park_choppers not working

### ISSUE THREAD

**Issue by Tom-Willemsen** [ISISComputingGroup/IBEX#4102]
As an instrument scientist on LET I need to be able to park my choppers reliably. Email from LET scientist states:

> This is another command specific to LET which would stop and park all the choppers on LET in SECI. The equivalent on IBEX does not work at all.

**Comment by kjwoodsISIS** [comment]
We need to understand why the IBEX "park choppers" script does not work.  Was this script tested during the migration?  If so, why was the problem not spotted then?

**Comment by Tom-Willemsen** [comment]
We tested park choppers with the LET scientists. The individual functions work correctly. The function to park all choppers simultaneously does not work. We have agreed to test this in the shutdown as it will be easiest to check this with real hardware.

**Comment by Tom-Willemsen** [comment]
We have been given permission to test things on LET from now until the end of the shutdown. Don't run chopper 5 at more than 10Hz if the small blockhouse is unlocked, otherwise we are free to test whatever we need.

**Comment by Tom-Willemsen** [comment]
Although they are not the same ticket, it makes sense to look at this alongside https://github.com/ISISComputingGroup/IBEX/issues/4123 as we will need access to LET's choppers to test.

**Comment by Tom-Willemsen** [comment]
It looks like this is caused by a heap corruption crashing Python itself. In theory this should not be possible from pure python code. I will investigate further exactly what problematic code we have used. It seems very reproducible on LET.

Call stack:

![image](https://user-images.githubusercontent.com/24375813/56586449-a9a44580-65d7-11e9-9717-039018c35cc9.png)

**Comment by Tom-Willemsen** [comment]
I had trouble reproducing this in a way consistent enough to produce a proper bug report. My hunch is that it is something in CaChannel that is not locking a structure appropriately, but I don't have enough good proof of this. I was able to reproduce various segfaults/heap corruptions, but it stopped crashing when I connected a debugger to the process!

Instead of chasing this issue too far, I have just added locking to LET's scripts. This seems to fix the issue with parking choppers in parallel on LET. I have created a separate ticket ( https://github.com/ISISComputingGroup/IBEX/issues/4276 ) to investigate further.

To review, we can test with real choppers on LET at the moment (check with me first), or may be able to replicate the config in simulation mode locally and test.
